tests/test_types.py

```diff
@@ -10,19 +11,31 @@
 testdata = {  # Test ID -> (Location args, expected string representation, sort order)
     # First arg must be a Path, or "<stdin>"
     "nothing": (("<stdin>",), "<stdin>", 111),
-    "abs_path": ((Path("/foo/bar"),), "/foo/bar", 211),
+    "abs_path": ((Path("/foo/bar"),), os.path.join(os.sep, "foo", "bar"), 211),
     "rel_path": ((Path("foo"),), "foo", 311),
     # Second arg refers to notebook cell, and is rendered in [square brackets]
     "no_path_cell": (("<stdin>", 1), "<stdin>[1]", 121),
-    "abs_path_cell": ((Path("/foo/bar"), 2), "/foo/bar[2]", 221),
+    "abs_path_cell": (
+        (Path("/foo/bar"), 2),
+        os.path.join(os.sep, "foo", "bar[2]"),
+        221,
+    ),
     "rel_path_cell": ((Path("foo"), 3), "foo[3]", 321),
     # Third arg is line number, and is prefixed by colon
     "no_path_cell_line": (("<stdin>", 1, 2), "<stdin>[1]:2", 122),
-    "abs_path_cell_line": ((Path("/foo/bar"), 2, 3), "/foo/bar[2]:3", 222),
+    "abs_path_cell_line": (
+        (Path("/foo/bar"), 2, 3),
+        os.path.join(os.sep, "foo", "bar[2]:3"),
+        222,
+    ),
     "rel_path_cell_line": ((Path("foo"), 3, 4), "foo[3]:4", 322),
     # Cell number is omitted for non-notebooks.
     "no_path_line": (("<stdin>", None, 2), "<stdin>:2", 112),
-    "abs_path_line": ((Path("/foo/bar"), None, 3), "/foo/bar:3", 212),
+    "abs_path_line": (
+        (Path("/foo/bar"), None, 3),
+        os.path.join(os.sep, "foo", "bar:3"),
+        212,
+    ),
```

Comment:
Hmm, I wonder if these tests require so rethinking? I guess absolute paths like `\foo\bar` makes sense on Windows? But are `os.path.join(os.sep, ...)` the appropriate way to create such paths? And should we also test paths with a drive prefix (protected by appropriate `pytest.skipif`)?

Reply:
> I guess absolute paths like \foo\bar makes sense on Windows? 

Yes. Both `\MyDir` and `C:\MyDir` make sense.

> But are os.path.join(os.sep, ...) the appropriate way to create such paths? 

Without `os.sep` we cannot access disc or system-level files, which work both for Windows and Linux systems.

>And should we also test paths with a drive prefix (protected by appropriate `pytest.skipif`)?

That is a good idea. I added it in 6df3f252ce58d29b1664c875438029b55815bc28

Reply:
I'm not 100% happy with how the second component of these test vectors (the string representation we expect to be produced for a given `Location` object) now depends on the output of `os.path.join()`, especially when the only thing we really want/expect it to do is the `/` -> `\` transformation. When I look at these test vectors now, it almost looks like we're testing `Path("foo/bar")` against `os.path.join("foo", "bar")`, which is not the point of these `Location` tests...

Another consideration is that we're passing extra ("almost special") characters to `os.path.join()` with statements like `os.path.join(os.sep, "foo", "bar[2]:3")`. The `[2]:3` suffix is not really part of the filename, and I have to _assume_ that `os.path.join()` passes them along unchanged (I would need to look up allowable characters in Windows filenames, and the implementation of `os.path.join()` to know for sure...)

In sum I think we're better off dropping `os.path.join()` in these test vectors, and instead doing something simpler like this in the test below:

```python
def test_location__str(args, string, _):
    assert str(Location(*args)) == string.replace("/", os.sep)
```

Reply:
I see your point. I changed the expected results to strings and used `tests_location__str` function as you proposed. Changes done in 5932b06faafcef4992d4469714805396634efbe9.